controllers/repository_controller.go

```diff
@@ -291,3 +295,31 @@ func (r RepositoryReconciler) ensureRatingServiceAccount(ctx context.Context, na
 
 	return nil
 }
+func (r *RepositoryReconciler) getRepoType(repo *corev1alpha1.Repository) (string, error) {
+	if !repo.IsOCI() {
+		url := repo.Spec.URL + "/api/charts"
+		req, err := http.NewRequest("GET", url, nil)
+		if err != nil {
+			return "", err
+		}
+		if repo.Spec.AuthSecret != "" {
+			username, password, _, _, _, err := corev1alpha1.ParseRepoSecret(r.Client, repo)
+			if err != nil {
+				return "", err
+			}
+			req.SetBasicAuth(username, password)
+		}
+		req.Header.Add("Content-Type", "application/json")
+
+		client := &http.Client{}
+		resp, err := client.Do(req)
+		if err != nil {
+			return "", err
+		}
+		if resp != nil && resp.StatusCode == http.StatusOK {
+			return string(corev1alpha1.RepositoryTypeChartmuseum), nil
+		}
+		defer resp.Body.Close()
```

Comment:
this defer func may never be called since you already return when it is `StatusOk`

Reply:
```golang

func (r *RepositoryReconciler) getRepoType(repo *corev1alpha1.Repository) (string, error) {
	var repoType string
       
	if repo.IsOCI() {
		return "oci", nil
	}

      
       
	return repoType, nil
}

func (repo *corev1alpha1.Repository)IsChartmuseum(url string,authSecret string) bool {
	req, err := http.NewRequest("GET", repo.Spec.URL + "/api/charts/kubebb-core", nil)
	if err != nil {
		return false
	}
	if repo.Spec.AuthSecret != "" {
		username, password, _, _, _, err := corev1alpha1.ParseRepoSecret(r.Client, repo)
		if err == nil {
		   req.SetBasicAuth(username, password)
		}
	}
	client := &http.Client{}
	resp, err := client.Do(req)
        if err != nil {
             return err
        }
       defer resp.Body.Close()
	if resp !== nil && resp.StatusCode != http.StatusNotFound {
	     return true
	 }
    return false
}
 
```

Reply:
@zqq454224016 @Abirdcfly @0xff-dev